### PR TITLE
`standardize_shape` normalizes the dimensions and tuple.

### PR DESCRIPTION
**Pull request by hertschuh**
- `standardize_shape` would allow int-like objects (e.g. NumPy scalars) to be in the shape, but would not resolve them to actual ints. This could cause issues later. With this change, objects that can be cast to int will be cast to int, which includes NumPy scalars, but also TensorFlow constant or eager tensors.
- `standardize_shape` would have custom code to handle `torch.Size`. Generalized it to turn anything iterable to a plain tuple.
- Added units tests
- Removed duplicate unit tests
- Added TensorFlow specific unit test

### CODE REVIEW

File: keras/src/backend/common/variables_test.py [hunk 1]
```diff
@@ -310,27 +310,64 @@ def test_name_validation(self):
...
+        ):
+            standardize_shape(shape_with_float)
+
+    def test_standardize_shape_with_object(self):
+        shape_with_float = (3, 4, object())
+        with self.assertRaisesRegex(
+            ValueError,
+            "Cannot convert .* to a shape. Found invalid dimension .*object",
+        ):
+            standardize_shape(shape_with_float)
```
[the remark this answers is not in the document]
Comment (hertschuh) [reply keras/src/backend/common/variables_test.py]: Fixed

File: keras/src/backend/common/variables_test.py [hunk 2]
```diff
@@ -310,27 +310,64 @@ def test_name_validation(self):
...
         ):
             standardize_shape(42)
 
     def test_standardize_shape_with_valid_input(self):
-        """Tests standardizing shape with valid input."""
+        shape = (3, 4, 5)
+        standardized_shape = standardize_shape(shape)
+        self.assertEqual(standardized_shape, (3, 4, 5))
+
+    def test_standardize_shape_with_valid_inputWith_none(self):
```
[the remark this answers is not in the document]
Comment (hertschuh) [reply keras/src/backend/common/variables_test.py]: Fixed
